### PR TITLE
Kiali 2779 Handle Jaeger connections with authentication

### PR DESCRIPTION
**Pull request by aljesusg**
JIRA: [KIALI-2779](https://issues.jboss.org/browse/KIALI-2779)

Require: 

1. https://github.com/kiali/kiali/pull/1129
2. https://github.com/kiali/kiali/pull/1120

Configmap

```yaml
  tracing:
    auth:
      ca_file: /var/run/secrets/kubernetes.io/serviceaccount/service-ca.crt
      insecure_skip_verify: true
      password: ''
      token: ''
      type: bearer
      use_kiali_token: true
      username: ''
    enabled: true
    namespace: istio-system
    service: ''
    url: ''
```

### CODE REVIEW

File: business/jaeger_helper.go [hunk 1]
```diff
@@ -4,6 +4,7 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
+	"github.com/kiali/kiali/util/httputil"
 	"io/ioutil"
```
Comment (aljesusg) [inline business/jaeger_helper.go]: I need to move this to the proper site

**Comment by aljesusg** [comment]
Oauth with Kiali and Jaeger
oauth_with_kiali

**Comment by aljesusg** [comment]
working
Screenshot from 2019-06-12 17-31-15

File: operator/roles/kiali-deploy/defaults/main.yml [hunk 2]
```diff
@@ -66,6 +66,14 @@ kiali_defaults:
       custom_metrics_url: ""
       url: ""
     tracing:
+      auth:
```
Comment (jotak) [inline operator/roles/kiali-deploy/defaults/main.yml]: don't forget to also update the kiali_cr.yaml file which stands as a documentation

Comment (aljesusg) [reply operator/roles/kiali-deploy/defaults/main.yml]: oh you're right

Comment (aljesusg) [reply operator/roles/kiali-deploy/defaults/main.yml]: @jotak  Did you try this ? I was checking and I didn't need to set the url https://jaeger-query.istio-system.svc to fetch the traces o.O and it's works...I only see this if jaeger is in another namespace

Comment (aljesusg) [reply operator/roles/kiali-deploy/defaults/main.yml]: For example for jaeger-query would be https://jaeger-query.istio-system.svc:16686 and health check  https://jaeger-query.istio-system.svc:16687

**Comment by hhovsepy** [comment]
cc @skondkar

**Comment by aljesusg** [comment]
@skondkar ping me if you need help in the IRC

**Pull request merged**